src/components/sectionList/translation/TranslationDialog.tsx

```diff
@@ -0,0 +1,86 @@
+import i18n from '@dhis2/d2-i18n'
+import {
+    Modal,
+    ModalContent,
+    ModalTitle,
+    SingleSelectField,
+    SingleSelectOption,
+} from '@dhis2/ui'
+import React, { useMemo } from 'react'
+import { BaseListModel } from '../../../lib'
+import { Loader } from '../../loading'
+import style from './translation.module.css'
+import { TranslationForm } from './TranslationForm'
+import { useDBLocales, useLastSelectedLocale } from './translationFormHooks'
+
+type TranslationDialogProps = {
+    onClose: () => void
+    model: BaseListModel
+}
+
+export const TranslationDialog = ({
+    onClose,
+    model,
+}: TranslationDialogProps) => {
+    const dbLocalesQuery = useDBLocales()
+    const dbLocales = dbLocalesQuery.data
+
+    const [selectedLocaleString, setSelectedLocaleString] =
+        useLastSelectedLocale()
+
+    const selectedLocale = useMemo(() => {
+        if (!selectedLocaleString || !dbLocalesQuery.data) {
+            return undefined
+        }
+        return dbLocalesQuery.data?.find(
+            (locale) => locale.locale === selectedLocaleString
+        )
+    }, [selectedLocaleString, dbLocales])
```

Comment:
Getting a eslint error here.

```suggestion
      const selectedLocale = useMemo(() => {
        if (!selectedLocaleString || !dbLocales) {
            return undefined
        }
        return dbLocales.find(
            (locale) => locale.locale === selectedLocaleString
        )
    }, [selectedLocaleString, dbLocales])
```